khmer/khmer_args.py

```diff
@@ -46,15 +43,61 @@ def build_hash_args(descr=None, epilog=None):
     parser.add_argument('--ksize', '-k', type=int, default=env_ksize,
                         help='k-mer size to use')
     parser.add_argument('--n_tables', '-N', type=int,
-                        default=env_n_tables,
+                        default=None,
                         help='number of k-mer counting tables to use')
     parser.add_argument('--min-tablesize', '-x', type=float,
-                        default=env_tablesize,
+                        default=None,
                         help='lower bound on tablesize to use')
 
+    parser.add_argument('--max-memory', metavar='max_memory',
+                        type=float, default=None, help='maximum memory to use')
+
     return parser
 
 
+def update_memory_parameters(parser, args):
+    env_n_tables = os.environ.get('KHMER_N_TABLES', DEFAULT_N_TABLES)
+    env_tablesize = os.environ.get('KHMER_MIN_TABLESIZE',
+                                   DEFAULT_MIN_TABLESIZE)
+
+    # can't allow all three! (a, b, c)
+    if args.n_tables and args.min_tablesize and args.max_memory:
+        raise exception("cannot specify all three.")  # @CTB
+
+    # if none are specified, set to defaults. (not a, not b, not c)
+    if not (args.n_tables or args.max_memory or args.min_tablesize):
+        args.n_tables = env_n_tables
+        args.min_tablesize = env_tablesize
+
+    # just tablesize and max memory (a, b, not c)
+    if args.n_tables is None and (args.max_memory and args.min_tablesize):
+        if parser.hashtype == 'counting':
+            tablesize = int(args.max_memory / float(args.min_tablesize))
+        elif parser.hashtype == 'hashbits':
+            tablesize = int(8 * args.max_memory / float(args.min_tablesize))
```

Comment:
this should be `args.n_tables = int(8 * args.max_memory / float(args.min_tablesize))` not tablesize